src/components/Table/hoc/withTableSettings/TableColumnSetup/DndTreeSelect.tsx

```diff
@@ -0,0 +1,139 @@
+    items,
+    setItems,
+    initialDroppableId,
+    renderContainer: propsRenderContainer = DEFAULT_RENDER_CONTAINER,
+    ...treeSelectNativeProps
+}: DndTreeSelectProps<T>) {
+    const uniqId = useUniqId();
+
+    const renderContainer = React.useCallback<RenderContainerType<DndTreeSelectItem>>(
+        ({renderItem, visibleFlattenIds, items: _items, containerRef, id}) => {
+            const handleDrugEnd: OnDragEndResponder = ({destination, source}) => {
+                if (destination?.index !== undefined && destination?.index !== source.index) {
+                    const newItemsOrder = reorderArray(items, source.index, destination.index);
+
+                    setItems(newItemsOrder);
+                }
+            };
+
+            const visibleFlattenItemList = visibleFlattenIds.map((visibleFlattenId, idx) =>
+                renderItem(visibleFlattenId, idx),
+            );
+
+            const renderList = () => (
+                <ListContainerView ref={containerRef} id={id}>
+                    <DragDropContext onDragEnd={handleDrugEnd}>
+                        <Droppable droppableId={initialDroppableId ?? uniqId}>
+                            {(droppableProvided) => {
+                                return (
+                                    <div
+                                        {...droppableProvided.droppableProps}
+                                        ref={droppableProvided.innerRef}
+                                    >
+                                        {visibleFlattenItemList}
+                                        {droppableProvided.placeholder}
+                                    </div>
+                                );
+                            }}
+                        </Droppable>
+                    </DragDropContext>
+                </ListContainerView>
+            );
+
+            return propsRenderContainer({renderList});
+        },
+        [items, setItems, propsRenderContainer],
+    );
+
+    const renderDndItem = React.useCallback<RenderItem<DndTreeSelectItem>>(
+        (item, state, _, idx) => {
+            const endSlot =
+                item.endSlot ?? (item.isDragDisabled ? undefined : <Icon data={Grip} size={16} />);
+
+            const commonProps = {
+                ...state,
+                ...item,
+                endSlot,
+            };
+
+            return (
+                <Draggable
+                    draggableId={state.id}
+                    index={Number(idx)}
+                    key={`item-key-${state.id}`}
+                    isDragDisabled={item.isDragDisabled}
+                >
+                    {(provided, snapshot) => {
+                        const style: React.CSSProperties = {
+                            ...provided.draggableProps.style,
+                        };
+
+                        // not expected offset appears, one way to fix - remove this offsets explicitly
```

Comment:
use `renderClone` prop in `Droppable` component to prevent this behaviour. It's officeal [recommended](https://stackoverflow.com/questions/54982182/react-beautiful-dnd-drag-out-of-position-problem) way to work with absolute positioning dnd elements in `react-beautiful-dnd` lib 

It's already implemented in dnd TreeSelect example in storybook

Reply:
why `renderClone` is recommended ? I have never saw this solution while searching, and i saw a lot of using `left, top` or `position`.

Even link you send does not have this solution 

<img width="1401" alt="image" src="https://github.com/gravity-ui/uikit/assets/43282255/ffc34e7c-2241-4549-8b52-b8843119e974">